roles/foreman_installer/tasks/main.yml

```
@@ -0,0 +1,28 @@
+---
+- name: Check that necessary variables are defined
+  assert:
+    that:
+      - foreman_installer_scenario is defined
+
+- name: Check if initial install has happened
+  stat:
+    path: /etc/foreman-installer/scenarios.d/.installed
+  register: installed
+
+- name: 'Run installer in noop mode'
+  command: >
+    {{ foreman_installer_command }}
+    --scenario {{ foreman_installer_scenario }}
+    --noop
+  register: foreman_installer_noop
+  changed_when: false
```

```suggestion
  changed_when: false
  failed_when: foreman_installer_noop.rc not in [0, 2]
```

Otherwise this `command` will be marked as failed when there are actually changes to be made and the installer exited with 2.